syntax = "proto3";

package envoy.extensions.tracers.opentelemetry.samplers.v3;

import "envoy/config/core/v3/http_uri.proto";

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.envoy.extensions.tracers.opentelemetry.samplers.v3";
option java_outer_classname = "DynatraceSamplerProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/tracers/opentelemetry/samplers/v3;samplersv3";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: Dynatrace Sampler config]
//
// Configuration for the Dynatrace Sampler extension.
//
// [#extension: envoy.tracers.opentelemetry.samplers.dynatrace]
// [#next-free-field: 6]
message DynatraceSamplerConfig {
  string tenant = 1;

  string cluster_id = 2;

  config.core.v3.HttpUri http_uri = 3;

  string token = 4;

  // Default number of root spans per minute allowed by the sampler.
  // The value is used if the number can't be requested from the Dynatrace cluster.
  uint32 root_spans_per_minute = 5;
}
